src/services/containerServices.ts

```diff
@@ -87,4 +87,24 @@ export class ECSContainer implements IContainerServices {
     }
     return '';
   }
+
+  async stopZombieECSTask(taskId: string) {
+    const clusterName = this._config.get<string>('taskDefinitionFamily');
+    try {
+      const res = await this._client.describeTasks({
+        cluster: clusterName,
+        tasks: [taskId],
+      });
+
+      // Only stop ECS task if it's still running
+      if (res.tasks?.[0].lastStatus === 'RUNNING') {
+        await this._client.stopTask({
```

Comment:
Nit/Non-blocking suggestion:

It might be valuable in the future to attempt to log the `task` object on the response if it exists - just for the sake of being able to diagnose if the task can't be stopped, or diagnosing why the task became hanging. This can definitely be addressed in a future ticket though!

https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-ecs/interfaces/stoptaskcommandoutput.html

Reply:
Good idea! Let's address this in a future ticket so we can test output, formatting, etc. separately.